sklearn/metrics/pairwise.py

```diff
@@ -1943,40 +1949,51 @@ def distance_metrics():
     return PAIRWISE_DISTANCE_FUNCTIONS
 
 
-def _dist_wrapper(dist_func, dist_matrix, slice_, *args, **kwargs):
+def _dist_wrapper(dist_func, dist_matrix, slice_, *args, transpose=False, **kwargs):
     """Write in-place to a slice of a distance matrix."""
-    dist_matrix[:, slice_] = dist_func(*args, **kwargs)
+    if transpose:
+        dist_matrix[slice_, ...] = dist_func(*args, **kwargs).T
+    else:
+        dist_matrix[..., slice_] = dist_func(*args, **kwargs)
```

Comment:
Codecov complains here and it is true that this line is never used, as we only use `_dist_wrapper` inside of `_parallel_pairwise`.

@ogrisel would you be happy for me to change the default behaviour of `_dist_wrapper` so that it always transposes?

Reply:
Done